android/festago/app/src/test/java/com/festago/festago/presentation/ui/home/festivallist/FestivalListViewModelTest.kt

```diff
@@ -0,0 +1,115 @@
+package com.festago.festago.presentation.ui.home.festivallist
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.festago.festago.domain.model.Festival
+import com.festago.festago.domain.repository.FestivalRepository
+import com.festago.festago.presentation.mapper.toPresentation
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.lang.Exception
+import java.time.LocalDate
+
+class FestivalListViewModelTest {
+    private lateinit var vm: FestivalListViewModel
+    private lateinit var festivalRepository: FestivalRepository
+
+    private val fakeFestivals = List(5) {
+        Festival(
+            it.toLong(),
+            "테코대학교 $it",
+            LocalDate.of(2023, 5, 15),
+            LocalDate.of(2023, 5, 19),
+            "",
+        )
+    }
+
+    @get:Rule
+    val instantExecutorRule = InstantTaskExecutorRule()
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Before
+    fun setUp() {
+        Dispatchers.setMain(UnconfinedTestDispatcher())
+        festivalRepository = mockk()
+        vm = FestivalListViewModel(festivalRepository)
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @After
+    fun finish() {
+        Dispatchers.resetMain()
+    }
+
+    @Test
+    fun `축제 목록 받아오기에 성공하면 성공 상태이고 축제 목록을 반환한다`() {
+        // given
+        coEvery {
+            festivalRepository.loadFestivals()
+        } answers {
+            Result.success(fakeFestivals)
+        }
+
+        // when
+        vm.loadFestivals()
+
+        // then
+        assertThat(vm.uiState.value).isInstanceOf(FestivalListUiState.Success::class.java)
+        assertThat(vm.uiState.value?.shouldShowSuccess).isEqualTo(true)
+        assertThat(vm.uiState.value?.shouldShowLoading).isEqualTo(false)
+        assertThat(vm.uiState.value?.shouldShowError).isEqualTo(false)
+
+        val actual = (vm.uiState.value as FestivalListUiState.Success).festivals
+        val expected = fakeFestivals.map { it.toPresentation() }
+        assertThat(actual).isEqualTo(expected)
```

Comment:
```suggestion
        // then
        assertThat(vm.uiState.value).isInstanceOf(FestivalListUiState.Success::class.java)
        
        // and
        assertThat(vm.uiState.value?.shouldShowSuccess).isEqualTo(true)
        assertThat(vm.uiState.value?.shouldShowLoading).isEqualTo(false)
        assertThat(vm.uiState.value?.shouldShowError).isEqualTo(false)

        // and
        val actual = (vm.uiState.value as FestivalListUiState.Success).festivals
        val expected = fakeFestivals.map { it.toPresentation() }
        assertThat(actual).isEqualTo(expected)
```

3가지 테스트들의 역할들이 다르다고 생각하는데 다음과 같이 분리하는게 어떤가요??

Reply:
좋은 생각입니다! 검증이 많으면 부연 설명도 같이 적어주면 더 가독성을 높일 수 있을 것 같아요ㅎㅎ

Reply:
반영했습니다! [6326a63](https://github.com/woowacourse-teams/2023-festa-go/pull/127/commits/6326a63953e084019f27e7760515c663c311d427)